fathom/src/core/semantics.rs

```diff
@@ -650,7 +662,13 @@ impl<'arena, 'env> ElimContext<'arena, 'env> {
                 // Otherwise call default with `head_expr`
                 let mut rigid_exprs = split.rigid_exprs.clone();
                 rigid_exprs.push(head_expr);
-                EvalContext::new(&mut rigid_exprs, self.flexible_exprs).eval(split.default_expr)
+                match split.default_expr {
+                    Some(default_expr) => {
+                        EvalContext::new(&mut rigid_exprs, self.flexible_exprs).eval(default_expr)
+                    }
+                    // FIXME: https://github.com/yeslogic/fathom/issues/340
+                    None => panic_any(Error::MissingConstDefault),
```

Comment:
I think it's fine to add the FIXME here - but when #340 is fixed this panic will probably remain - this should be caught by the elaborator when compiling the pattern matches.